File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Client/src/models/delegatedAccess/delegatedAccess.ts
import { Guid } from 'guid-typescript';

export class ImpersonationContext {
    public id: Guid;
    public requesterEmail: string;
    public recipientEmail: string;
    public isImpersonationContext: boolean;
    public isApproved: boolean;
    public data: Uint8Array;
    public identifierColumn: string;
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;

    constructor(impersonationContext: any) {
        this.id = impersonationContext.id ? Guid.parse(impersonationContext.id) : Guid.parse(Guid.EMPTY);
        this.requesterEmail = impersonationContext.requesterEmail || "";
        this.recipientEmail = impersonationContext.recipientEmail || "";
        this.isImpersonationContext = impersonationContext.isImpersonationContext || false;
        this.isApproved = impersonationContext.isApproved || false;
        this.data = impersonationContext.data || 0;
        this.identifierColumn = impersonationContext.identifierColumn || ""; 
        this.createdDate = impersonationContext.createdDate ? new Date(impersonationContext.createdDate) : undefined;
        this.createdBy = impersonationContext.createdBy || "";
        this.updatedDate = impersonationContext.updatedDate ? new Date(impersonationContext.updatedDate) : undefined;
        this.updatedBy = impersonationContext.updatedBy || "";
    }
}

File: ISL.ReIdentification.Core/Models/Foundations/DelegatedAccesses/DelegatedAccess.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Core.Models.Foundations.DelegatedAccesses
{
    public class DelegatedAccess : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string RequesterFirstName { get; set; } = string.Empty;
        public string RequesterLastName { get; set; } = string.Empty;
        public string RequesterEmail { get; set; } = string.Empty;
        public string RecipientFirstName { get; set; } = string.Empty;
        public string RecipientLastName { get; set; } = string.Empty;
        public string RecipientEmail { get; set; } = string.Empty;
        public bool IsSystemAccess { get; set; }
        public string Reason { get; set; } = string.Empty;
        public string ReasonMessage { get; set; } = string.Empty;
        public bool? IsApproved { get; set; } = null;
        public byte[]? Data { get; set; }
        public string IdentifierColumn { get; set; } = string.Empty;
        public string CreatedBy { get; set; } = string.Empty;
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
    }
}